### PR TITLE
LLaVA: latency issues

### PR DESCRIPTION
**Pull request by zucchini-nlp**
# What does this PR do?

Fixes latency issues in some llava models caused by the recent processing logic refactoring. It was reported by one of the users that the new release is twice slower than the prev release. The problem was in indentation where images were embedded in every run instead of only the pre-fill stage

I believe it is worth fixing and merging until https://github.com/huggingface/transformers/pull/33424 is merged to unblock e from updating hub configs and enabling the new logic

### CODE REVIEW

File: src/transformers/models/llava_next_video/modeling_llava_next_video.py [hunk 1]
```diff
@@ -1110,17 +1110,6 @@ def prepare_inputs_for_generation(
...
-        if input_ids is not None:
-            img_token_not_enough = (input_ids == self.config.image_token_index).sum(
-                1
-            ).max() < self.config.image_seq_length
-            video_token_not_enough = (input_ids == self.config.video_token_index).sum(
-                1
-            ).max() < self.config.video_seq_length
-            legacy_processing = (img_token_not_enough and pixel_values is not None) or (
-                video_token_not_enough and pixel_values_videos is not None
-            )
```
Comment (ArthurZucker) [inline src/transformers/models/llava_next_video/modeling_llava_next_video.py:1122]: glad that this is removed, never saw this being added (was not pinged on the PR!)

**Review by ArthurZucker — APPROVED** [review]
Thanks, noting that this is needed in a patch 🤗

**Comment by jiqing-feng** [comment]
Hi @zucchini-nlp @ArthurZucker . This PR could lead the issue #34696 . Do you mind take a look? Thanks.

**Comment by zucchini-nlp** [comment]
@jiqing-feng I'll be updating all model configs starting next week to enable the new logic for processing. For the linked issue, will take a look soon , thanks

**Pull request merged**